### PR TITLE
Add yes/no confirmation to delete-cluster command.

### PR DESCRIPTION
**Pull request by shane-containership**
Also add optional '-y' argument, to disable promt, for non-interactive use

### CODE REVIEW

File: lib/cli-v2.js [hunk 1]
```diff
@@ -639,39 +639,63 @@ module.exports.commands.push({
...
+                        method: 'DELETE',
+                        json: true,
+                        headers: {
+                            'authorization': `Bearer ${token}`
+                        }
+                    };
 
-                    return console.info('Successfully deleted the cluster!');
-                });
+                    return request(request_options, function(err, response) {
```
Comment (normanjoyner) [inline lib/cli-v2.js]: arrow function here

File: lib/cli-v2.js [hunk 2]
```diff
@@ -639,39 +639,63 @@ module.exports.commands.push({
...
+
+                prmpt.get({
+                    name: 'value',
+                    message: chalk.red(`Are you sure you want to delete cluster: ${argv.cluster_id} ? THIS OPERATION CANNOT BE REVERSED!`),
+                    validator: /y[es]*|n[o]?/,
+                    warning: 'Must respond yes or no',
+                    default: 'no',
+                    ask: () => {
+                        // only prompt for confirmation if '-y' flag not passed in CLI
+                        return!argv.y;
```
Comment (normanjoyner) [inline lib/cli-v2.js]: can we change this to a `-f` (aliased to `--force`) option? i think that may be a bit more clear as an option passed to the command from a ux perspective, as opposed to `--yes`

File: lib/cli-v2.js [hunk 3]
```diff
@@ -639,39 +639,63 @@ module.exports.commands.push({
...
-                return request(request_options, function(err, response) {
+                }, (err, confirmation) => {
                     if(err) {
-                        return console.error('There was an error deleting the cluster!');
+                        return console.error(err.message || 'There was an error parsing your response!');
                     }
+                    if(argv.y || (confirmation.value === 'yes')) {
+                        return executeDelete(org_id, argv);
+                    }
+                    return console.error('Delete Cluster command aborted!');
```
Comment (normanjoyner) [inline lib/cli-v2.js]: lowercase cluster

File: lib/cli-v2.js [hunk 4]
```diff
@@ -639,39 +639,63 @@ module.exports.commands.push({
...
                     }
-                };
-
-                return request(request_options, function(err, response) {
+                }, (err, confirmation) => {
                     if(err) {
-                        return console.error('There was an error deleting the cluster!');
+                        return console.error(err.message || 'There was an error parsing your response!');
                     }
+                    if(argv.y || (confirmation.value === 'yes')) {
```
Comment (normanjoyner) [inline lib/cli-v2.js]: reflect new option here

**Comment by shane-containership** [comment]
@normanjoyner , I added the changes above . Please let me know if there is a better way to set the 'f' alias option.

File: lib/cli-v2.js [hunk 5]
```diff
@@ -639,39 +639,67 @@ module.exports.commands.push({
...
+
+                prmpt.get({
+                    name: 'value',
+                    message: chalk.red(`Are you sure you want to delete cluster: ${argv.cluster_id} ? THIS OPERATION CANNOT BE REVERSED!`),
+                    validator: /y[es]*|n[o]?/,
+                    warning: 'Must respond yes or no',
+                    default: 'no',
+                    ask: () => {
+                        // only prompt for confirmation if '-f', or '--force' flag not passed in CLI
+                        let flagged_args = _.omit(argv, ['h', 'help', '$0', '_', 'cluster_id']);
```
Comment (normanjoyner) [inline lib/cli-v2.js]: if you add a top level `options` object like the following:
```
options: {
    force: {
        description: 'Force cluster deletion without prompt',
        alias: 'f',
        type: 'boolean',
        default: false
    }
}
```

you can remove lines 654-657, and just leave `return !argv.force;`

**Comment by shane-containership** [comment]
@normanjoyner , I made the proposed changes above. This should be ready to merge.

File: lib/cli-v2.js [hunk 6]
```diff
@@ -635,43 +635,75 @@ module.exports.commands.push({
...
+
+                prmpt.get({
+                    name: 'value',
+                    message: chalk.red(`Are you sure you want to delete cluster: ${argv.cluster_id} ? THIS OPERATION CANNOT BE REVERSED!`),
+                    validator: /y[es]*|n[o]?/,
+                    warning: 'Must respond yes or no',
+                    default: 'no',
+                    ask: () => {
+                        // only prompt for confirmation if '-f', or '--force' flag not passed in CLI
+                        return!argv.force;
```
Comment (normanjoyner) [inline lib/cli-v2.js]: add space after `return`

File: lib/cli-v2.js [hunk 7]
```diff
@@ -805,3 +837,4 @@ module.exports.commands.push({
         }
     ]
 });
+
```
Comment (normanjoyner) [inline lib/cli-v2.js]: remove whitespace

File: lib/cli-v2.js [hunk 8]
```diff
@@ -635,43 +635,75 @@ module.exports.commands.push({
...
-                        return console.error('There was an error deleting the cluster!');
+                    if(!org_id) {
+                        return console.error(`You must have an active_organization to edit a cluster. See '${argv.$0} cloud organization use <org_id>'.`);
                     }
+                    const request_options = {
+                        url: `${constants.environment.CLOUD_API_BASE_URL}/v2/organizations/${org_id}/clusters/${argv.cluster_id}`,
+                        method: 'DELETE',
+                        json: true,
+                        headers: {
+                            'authorization': `Bearer ${token}`
```
Comment (normanjoyner) [inline lib/cli-v2.js]: can you capitalize `authorization`, and remove single quotes

File: lib/cli-v2.js [hunk 9]
```diff
@@ -635,43 +635,75 @@ module.exports.commands.push({
         {
             name: 'delete <cluster_id>',
             description: 'Delete Containership cloud cluster.',
+            options: {
+                force: {
+                    description: 'Force cluster deletion without prompt',
```
Comment (normanjoyner) [inline lib/cli-v2.js]: nit: punctuation after `prompt`

File: lib/cli-v2.js [hunk 10]
```diff
@@ -635,43 +635,75 @@ module.exports.commands.push({
...
-                        'authorization': `Bearer ${token}`
+                prmpt.message = '';
+                prmpt.delimiter = chalk.red(':');
+                prmpt.start();
+
+                prmpt.get({
+                    name: 'value',
+                    message: chalk.red(`Are you sure you want to delete cluster: ${argv.cluster_id} ? THIS OPERATION CANNOT BE REVERSED!`),
+                    validator: /y[es]*|n[o]?/,
+                    warning: 'Must respond yes or no',
```
Comment (normanjoyner) [inline lib/cli-v2.js]: it doesn't appear this warning ever gets printed. can we make it print as there is no indication what response is valid (yes or no)

File: lib/cli-v2.js [hunk 11]
```diff
@@ -635,43 +635,75 @@ module.exports.commands.push({
...
-                    json: true,
-                    headers: {
-                        'authorization': `Bearer ${token}`
+                prmpt.message = '';
+                prmpt.delimiter = chalk.red(':');
+                prmpt.start();
+
+                prmpt.get({
+                    name: 'value',
+                    message: chalk.red(`Are you sure you want to delete cluster: ${argv.cluster_id} ? THIS OPERATION CANNOT BE REVERSED!`),
```
Comment (normanjoyner) [inline lib/cli-v2.js]: nit: can you remove the space before the question mark

**Comment by normanjoyner** [comment]
@shane-containership can you please resolve merge conflicts

**Pull request merged**